### PR TITLE
19th issuev1 autoblock blogpages

### LINKED ISSUE #19: Issue - 19 - Configuring blog introduction with values from metadata
image

image

### PR DESCRIPTION
**Pull request by meejain**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #19 

Test URLs:
- Before: https://main--impala--mkclawson.hlx.live/
- After: https://19th-issuev1-autoblock-blogpages--impala--mkclawson.hlx.live/blog/rocky-mountain-snow

### CODE REVIEW

**Comment by meejain** [comment]
Auto Blocking is working. Will be working on fonts, placement & lint issues tomorrow.

File: scripts/scripts.js [hunk 1]
```diff
@@ -31,6 +32,31 @@ function buildHeroBlock(main) {
...
+    const authorcontent = getMetadata('author');
+    const publishedcontent = getMetadata('published');
+    const readcontent = getMetadata('readtime');
+    const section = document.createElement('div');
+    const elems = [authorcontent, publishedcontent, readcontent];
+    const subsection1 = document.createElement('div');
+    elems.forEach((x, i) => { const col = document.createElement('div'); col.textContent = x; subsection1.appendChild(col); });
+    section.appendChild(subsection1);
+    subsection1.after(h1);
+    const updatedcontent = getMetadata('updated');
```
Comment (helms-charity) [inline scripts/scripts.js:48]: There's no metadata for "updated", remember -- this is going to just be a manual text addition to the body copy if the author decides to use it. Only the published date is autoblocked.

File: scripts/scripts.js [hunk 2]
```diff
@@ -43,6 +69,12 @@ function buildAutoBlocks(main) {
     // eslint-disable-next-line no-console
     console.error('Auto Blocking failed', error);
```
Comment (helms-charity) [inline scripts/scripts.js:70]: You are missing some important things here.
Line 70 should look like this:

function buildAutoBlocks(main) {
  try {
    buildAutoblogBlock(main);
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error('Auto Blocking failed', error);
  }

File: scripts/scripts.js [hunk 3]
```diff
@@ -31,6 +32,31 @@ function buildHeroBlock(main) {
   }
 }
 
+function buildautoblogBlock(main) {
```
Comment (helms-charity) [inline scripts/scripts.js:35]: We use camel case for functions, so please change  all to "buildAutoblogBlock"

**Review by helms-charity — CHANGES_REQUESTED** [review]

File: scripts/scripts.js (hunk 2 above)
Comment (meejain) [reply scripts/scripts.js:70]: @helms-charity  - not sure what important thing u r referring here. Is there anything apart from Camel Casing that you are referring here? The Code is already within the function which is within the function buildAutoBlocks(main) function. 
 
function buildAutoBlocks(main) {
  try {
    buildHeroBlock(main);
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error('Auto Blocking failed', error);
  }
  try {
    buildAutoblogBlock(main);
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error('Auto Blocking failed', error);
  }
}

File: scripts/scripts.js (hunk 3 above)
Comment (meejain) [reply scripts/scripts.js]: Done

File: scripts/scripts.js (hunk 2 above)
Comment (helms-charity) [reply scripts/scripts.js:69]: You are absolutely right to be confused -- I neglected to notice that Github had collapsed part of the code, O thought it was missing! My bad!!

**Pull request merged**